### PR TITLE
GPU: Add SDL_CalculateGPUTextureFormatSize

### PR DESCRIPTION
**Pull request by thatcosmonaut**
This is a fairly common client need and it's annoyingly nuanced to implement, so we should expose it.

### CODE REVIEW

**Review by flibitijibibo — APPROVED** [review]
If not as an export, maybe as part of documentation for the blocksize functions?

File: include/SDL3/SDL_gpu.h [hunk 1]
```diff
@@ -3715,6 +3715,23 @@ extern SDL_DECLSPEC bool SDLCALL SDL_GPUTextureSupportsSampleCount(
...
+ * \param h height in pixels.
+ * \param d depth in pixels.
+ * \returns the size of a texture with this format and dimensions.
+ *
+ * \since This function is available since SDL 3.2.0.
+ */
+extern SDL_DECLSPEC Uint32 SDLCALL SDL_CalculateGPUTextureFormatSize(
+    SDL_GPUTextureFormat format,
+    Uint32 w,
+    Uint32 h,
```
Comment (slouken) [inline include/SDL3/SDL_gpu.h:3732]:
```suggestion
    Uint32 height,
```

File: include/SDL3/SDL_gpu.h [hunk 2]
```diff
@@ -3715,6 +3715,23 @@ extern SDL_DECLSPEC bool SDLCALL SDL_GPUTextureSupportsSampleCount(
...
+    Uint32 d);
```
Comment (slouken) [inline include/SDL3/SDL_gpu.h:3733]:
```suggestion
    Uint32 depth);
```
It's 2024, we can afford extra characters for clarity. :)

File: src/gpu/SDL_gpu.c [hunk 3]
```diff
@@ -2796,3 +2796,16 @@ void SDL_ReleaseGPUFence(
...
 }
+
+Uint32 SDL_CalculateGPUTextureFormatSize(
+    SDL_GPUTextureFormat format,
+    Uint32 w,
+    Uint32 h,
+    Uint32 d)
+{
+    Uint32 pixelRowsPerBlock = Texture_GetBlockSize(format);
+    Uint32 pixelColumnsPerBlock = pixelRowsPerBlock;
```
Comment (slime73) [inline src/gpu/SDL_gpu.c:2807]: Is this accurate for non-square block sizes?

File: include/SDL3/SDL_gpu.h [hunk 4]
```diff
@@ -3715,6 +3715,23 @@ extern SDL_DECLSPEC bool SDLCALL SDL_GPUTextureSupportsSampleCount(
...
+/**
+ * Calculate the size in bytes of a texture format with dimensions.
+ *
+ * \param format a texture format.
+ * \param width width in pixels.
+ * \param height height in pixels.
+ * \param depth depth in pixels.
+ * \returns the size of a texture with this format and dimensions.
+ *
+ * \since This function is available since SDL 3.2.0.
```
[the remark this answers is not in the document]
Comment (sezero) [reply include/SDL3/SDL_gpu.h]: wiki is reverting those back to 3.0.0, btw...

[the remark this answers is not in the document]
Comment (slouken) [reply include/SDL3/SDL_gpu.h]: Thanks for the heads up.

File: src/gpu/SDL_gpu.c (hunk 3 above)
Comment (thatcosmonaut) [reply src/gpu/SDL_gpu.c:2807]: You're right, we need to restructure this for ASTC.

**Comment by thatcosmonaut** [comment]
Addressed all the comments, let me know if there's anything else.

**Review by slouken — APPROVED** [review]

**Pull request merged**